tests/feature_resolution_editions.js

```diff
@@ -0,0 +1,300 @@
+        string string_val = 1;
+        string string_repeated = 2 [features.enum_type = CLOSED];
+
+        message Nested {
+            option features.(abc).d_e = deeply_nested_true;
+            option features.field_presence = IMPLICIT;
+            int64 count = 9;
+        }
+        
+        enum SomeEnum {
+            ONE = 1 [features.repeated_field_encoding = EXPANDED];
+            TWO = 2;
+        }
+    }`).root
+    rootEditionsOverriden.resolveAll();
+
+    // Should flip enum_type from default setting, inherit from Message,
+    // and keep everything else
+    test.same(rootEditionsOverriden.lookup("Message").fields.stringRepeated._features, {
+        enum_type: 'CLOSED',
+        field_presence: 'EXPLICIT',
+        json_format: 'LEGACY_BEST_EFFORT',
+        message_encoding: 'LENGTH_PREFIXED',
+        repeated_field_encoding: 'PACKED',
+        utf8_validation: 'VERIFY',
+        '(abc)': { d_e: 'deeply_nested_false' }
+      })
+
+    // Should inherit from default, and Message, only change field_presence and the custom extension
+    test.same(rootEditionsOverriden.lookup("Message").lookup("Nested")._features, {
+        enum_type: 'OPEN',
+        field_presence: 'IMPLICIT',
+        json_format: 'LEGACY_BEST_EFFORT',
+        message_encoding: 'LENGTH_PREFIXED',
+        repeated_field_encoding: 'PACKED',
+        utf8_validation: 'VERIFY',
+        '(abc)': { d_e: 'deeply_nested_true' } 
+      })
+
+    test.same(rootEditionsOverriden.lookup("Message").lookup("Nested")._features, {
+        enum_type: 'OPEN',
+        field_presence: 'IMPLICIT',
+        json_format: 'LEGACY_BEST_EFFORT',
+        message_encoding: 'LENGTH_PREFIXED',
+        repeated_field_encoding: 'PACKED',
+        utf8_validation: 'VERIFY',
+        '(abc)': { d_e: 'deeply_nested_true' } 
+      })
+
+    test.same(rootEditionsOverriden.lookupEnum("SomeEnum")._valuesFeatures["ONE"], {
+        enum_type: 'OPEN',
+        field_presence: 'EXPLICIT',
+        json_format: 'LEGACY_BEST_EFFORT',
+        message_encoding: 'LENGTH_PREFIXED',
+        repeated_field_encoding: 'EXPANDED',
+        utf8_validation: 'VERIFY',
+        '(abc)': { d_e: 'deeply_nested_false' } 
+    })
+
+    test.same(rootEditionsOverriden.lookupEnum("SomeEnum")._valuesFeatures["TWO"], {
+        enum_type: 'OPEN',
+        field_presence: 'EXPLICIT',
+        json_format: 'LEGACY_BEST_EFFORT',
+        message_encoding: 'LENGTH_PREFIXED',
+        repeated_field_encoding: 'PACKED',
+        utf8_validation: 'VERIFY',
+        '(abc)': { d_e: 'deeply_nested_false' } 
+    })
+
+    test.end();
+})
+// Tests precedence for different levels of feature resolution
+tape.test("feature resolution editions precedence", function(test) {
+    var root1 = protobuf.parse(`edition = "2023";
+    option features.amazing_feature = A;`).root.resolveAll()
+    
+    test.same(root1._features.amazing_feature, 'A');
+    
+    var root2 = protobuf.parse(`edition = "2023";
```

Comment:
Similar comment here, instead of bundling these all into one test with a bunch of rootN variables, we can keep them more isolated in their own test

Reply:
I don't think it's worth further splitting up this file into different tests given the amount of time I've already spent breaking this test up in each iteration of these comments (already 4 revisions to this test structure alone). It seems like you have a very specific idea for what you have these tests to look like, I'd encourage a commit if you have something very specific in mind.

Reply:
Your latest commit seems to be going in exactly the direction I was thinking.  "feature resolution inheritance file to oneofs" tests all of (defaults, inheritance from parent, overriding parent) for file -> oneof.  You *could* take it one step further and split that into 3 separate tests, but that seems excessive.  The key point I was trying to make was that the .proto file being tested should be minimal to make the expected behavior obvious (which you've accomplished above)

Reply:
Correction: that one actually should set `LEGACY_BEST_EFFORT` at the *message* level, not the file level.  Oneofs should inherit directly from their containing message not the file (file->message tests that part, no need to test file->oneof)

Reply:
This test doesn't use LEGACY_BEST_EFFORT.